lms/djangoapps/course_home_api/course_metadata/v1/views.py

```diff
@@ -65,8 +66,14 @@ def get(self, request, *args, **kwargs):
         )
 
         course = course_detail(request, request.user.username, course_key)
-        user_is_enrolled = CourseEnrollment.is_enrolled(request.user, course_key_string)
-
+        enrollment = CourseEnrollment.get_enrollment(request.user, course_key_string, select_related=['celebration'])
+        user_is_enrolled = enrollment is not None
+        should_celebrate_streak = False
+        if not is_masquerading_as_specific_student(request.user, course_key):
```

Comment:
Should this just be checking any sort of masquerading? Like if I'm masquerading as general learner or something, do we need to update in that case?

Reply:
I guess I'm just curious what happens if the staff user is masquerading as staff or as a full-access learner or verified learner. Does the code gracefully handle those situations as well?

Reply:
it should just be treating the accesses as accesses for the user in question unless they masquerade as a specific user, which i think is fine? not sure if that counts as graceful lol